x-pack/plugins/enterprise_search/public/applications/workplace_search/views/content_sources/components/synchronization/blocked_window_item.tsx

```diff
@@ -0,0 +1,157 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React from 'react';
+
+import {
+  EuiButton,
+  EuiComboBox,
+  EuiComboBoxOptionOption,
+  EuiDatePicker,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiSpacer,
+  EuiSuperSelect,
+  EuiText,
+} from '@elastic/eui';
+
+import { DAYS_OF_WEEK_LABELS } from '../../../../../shared/constants';
+import { BLOCK_LABEL, BETWEEN_LABEL, EVERY_LABEL, AND, REMOVE_BUTTON } from '../../../../constants';
+import { BlockedWindow, DAYS_OF_WEEK_VALUES } from '../../../../types';
+
+import {
+  FULL_SYNC_LABEL,
+  INCREMENTAL_SYNC_LABEL,
+  DELETION_SYNC_LABEL,
+  PERMISSIONS_SYNC_LABEL,
+  FULL_SYNC_DESCRIPTION,
+  INCREMENTAL_SYNC_DESCRIPTION,
+  DELETION_SYNC_DESCRIPTION,
+  PERMISSIONS_SYNC_DESCRIPTION,
+} from '../../constants';
+
+interface Props {
+  blockedWindow: BlockedWindow;
+}
+
+const syncOptions = [
+  {
+    value: 'full',
+    inputDisplay: FULL_SYNC_LABEL,
+    dropdownDisplay: (
+      <>
+        <strong>{FULL_SYNC_LABEL}</strong>
+        <EuiText size="s">{FULL_SYNC_DESCRIPTION}</EuiText>
+      </>
+    ),
+  },
+  {
+    value: 'incremental',
+    inputDisplay: INCREMENTAL_SYNC_LABEL,
+    dropdownDisplay: (
+      <>
+        <strong>{INCREMENTAL_SYNC_LABEL}</strong>
+        <EuiText size="s">{INCREMENTAL_SYNC_DESCRIPTION}</EuiText>
+      </>
+    ),
+  },
+  {
+    value: 'deletion',
+    inputDisplay: DELETION_SYNC_LABEL,
+    dropdownDisplay: (
+      <>
+        <strong>{DELETION_SYNC_LABEL}</strong>
+        <EuiText size="s">{DELETION_SYNC_DESCRIPTION}</EuiText>
+      </>
+    ),
+  },
+  {
+    value: 'permissions',
+    inputDisplay: PERMISSIONS_SYNC_LABEL,
+    dropdownDisplay: (
+      <>
+        <strong>{PERMISSIONS_SYNC_LABEL}</strong>
+        <EuiText size="s">{PERMISSIONS_SYNC_DESCRIPTION}</EuiText>
+      </>
+    ),
+  },
+];
+
+const dayPickerOptions = DAYS_OF_WEEK_VALUES.reduce((options, day) => {
+  options.push({
+    label: DAYS_OF_WEEK_LABELS[day.toUpperCase() as keyof typeof DAYS_OF_WEEK_LABELS],
+    value: day,
+  });
+  return options;
+}, [] as Array<EuiComboBoxOptionOption<string>>);
```

Comment:
Is it possible to use `map` here? I find it easier to understand than `reduce`.
If not possible feel free to resolve this comment.